wp-content/mu-plugins/pantheon/pantheon-plugin-install-notice.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * If a Pantheon site is in Git mode, hide the Plugin installation functionality and show a notice.
+ */
+
+if ( ! wp_is_writable( WP_PLUGIN_DIR ) ) {
+    if ( ! defined( 'DISALLOW_FILE_MODS' ) ) {
+        define( 'DISALLOW_FILE_MODS', true );
+    }
+
+    add_action( 'admin_notices', '_pantheon_plugin_install_notice' );
+}
+
+function _pantheon_plugin_install_notice() {
+    $screen = get_current_screen(); 
+    // Only show this notice on the plugins page.
+    if ( 'plugins' === $screen->id ) { ?>
```

Comment:
In case on a multisite install:

```suggestion
    if ( 'plugins' === $screen->id || 'plugins-network' === $screen->id ) { ?>
```